crates/core/src/factories/core_publish.rs

```diff
@@ -0,0 +1,247 @@
+//! Publish is a Kitsune2 module for publishing new ops and agent infos to peers.
+//!
+//! It consists of multiple parts:
+//! - A task that sends Op ids to peers
+//! - A task that adds received Op ids from other peers to the fetch queue
+//!
+//! ### Publish Ops
+//!
+//! #### Outgoing publishing of Ops
+//!
+//! A channel acts as the queue for outgoing ops to be published. Ops to be
+//! published are sent one by one through the channel to the receiving task,
+//! processing requests in incoming order and dispatching it to the
+//! transport module.
+//!
+//! #### Incoming of published Ops task
+//!
+//! Similarly to outgoing ops to be published, a channel serves as a queue
+//! for incoming published ops. The queue processes items in order of the
+//! incoming messages and adds the associated Op ids to the fetch queue.
+//!
+
+use kitsune2_api::*;
+use message_handler::PublishMessageHandler;
+use std::sync::Arc;
+use tokio::{
+    sync::mpsc::{channel, Receiver, Sender},
+    task::AbortHandle,
+};
+
+#[cfg(test)]
+mod test;
+
+mod message_handler;
```

Comment:
```suggestion
mod message_handler;

#[cfg(test)]
mod test;
```

Really minor, I prefer to find production modules before test ones

Reply:
addressed this one - resolving conversation because I adopted the change 1:1.